[FX] Further sanitize generated names

jamesr66a opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#44808 [FX] Further sanitize generated names**

Closes https://github.com/pytorch/pytorch/pull/44808

Expands the sanitization logic for generated names. In particular this expands the set of invalid chars we delete (e.g. `-`) and makes it so that a name that begins with a digit has a `_` prepended to it

Differential Revision: [D23739413](https://our.internmc.facebook.com/intern/diff/D23739413)

Commit:
[FX] Further sanitize generated names

[ghstack-poisoned]

dzhulgakov commented on torch/fx/graph.py:
```diff
@@ -140,8 +141,11 @@ def _name(self, target: Target) -> str:
         op = op.replace('.', '_')
-        op = op.replace('*', '')
+        # delete all characters that are illegal in a Python identifier
+        op = re.sub('[^0-9a-zA-Z_]', '', op)
```
maybe instead of deleting - replace with `_`. E.g. if someone creates `foo/bar` as a name, `foo_bar` is better replacement than `foobar`. Maybe `re.sub('[^0-9a-zA-Z_]+', '_', op)` to replace a sequence of banned chars with one

Commit:
Update on "[FX] Further sanitize generated names"


Closes https://github.com/pytorch/pytorch/pull/44808

Expands the sanitization logic for generated names. In particular this expands the set of invalid chars we delete (e.g. `-`) and makes it so that a name that begins with a digit has a `_` prepended to it

Differential Revision: [D23739413](https://our.internmc.facebook.com/intern/diff/D23739413)

[ghstack-poisoned]
